packages/connection-storage/src/connection-storage.ts

```diff
@@ -10,17 +10,56 @@ import {
   mergeSecrets,
   extractSecrets,
 } from './connection-secrets';
-import { getKeytarServiceName, deleteCompassAppNameParam } from './utils';
+import {
+  deleteCompassAppNameParam,
+  getKeytarServiceName,
+  parseStoredPassword,
+  throwIfAborted,
+} from './utils';
+import { ipcExpose, ipcInvoke } from '@mongodb-js/compass-utils';
+import {
+  serializeConnections,
+  deserializeConnections,
+} from './import-export-connection';
+import type {
+  ImportConnectionOptions,
+  ExportConnectionOptions,
+} from './import-export-connection';
 
 const { log, mongoLogId } = createLoggerAndTelemetry('CONNECTION-STORAGE');
 
-export class ConnectionStorage {
+export class ConnectionStorageMain {
+  private readonly path!: string;
   private readonly folder = 'Connections';
-  private readonly keytarServiceName: string;
   private readonly maxAllowedRecentConnections = 10;
+  private readonly keytarServiceName = getKeytarServiceName();
+
+  private static instance: ConnectionStorageMain | null;
+
+  constructor(path = '') {
```

Comment:
Not that we have a documented pattern around this but classes we use in the main process are usually completely static, I think it's fine either way whether we want static classes or just singletons created with a constructor, but I wouldn't mix two approaches in one

Reply:
fixed in [eb01bda](https://github.com/mongodb-js/compass/pull/4704/commits/eb01bda830cf988d8fe568bf267f9240f00c9a89)